LayoutTests/imported/w3c/web-platform-tests/screen-orientation/fullscreen-interactions.html

```diff
@@ -5,29 +5,85 @@
 
   promise_test(async (t) => {
+    t.add_cleanup(makeCleanup());
     await test_driver.bless("request full screen");
     await document.documentElement.requestFullscreen();
-    const lockPromise = screen.orientation.lock(getOppositeOrientation());
-    await document.exitFullscreen();
-    await promise_rejects_dom(t, "AbortError", lockPromise);
-  }, "Fully unlocking the screen orientation causes a pending lock to be aborted");
+    const currentOrientation = screen.orientation.type;
+    await screen.orientation.lock(
+      getOppositeOrientation()
+    );
+  }, "fullscreen and orientation support");
 
   promise_test(async (t) => {
     const iframe = await attachIframe();
+    t.add_cleanup(makeCleanup(iframe));
+    const iframeWindow = iframe.contentWindow;
+    await test_driver.bless("request full screen");
+    await document.documentElement.requestFullscreen();
+    const currentOrientation = window.screen.orientation.type;
+    const lockPromise = iframeWindow.screen.orientation.lock(
+      getOppositeOrientation()
+    );
+    const fsExitPromise = document.exitFullscreen();
+    await promise_rejects_dom(
+      t,
+      "SecurityError",
+      iframeWindow.DOMException,
+      lockPromise
+    );
+    await fsExitPromise;
+  }, "Iframe can't itself know if it's parent is fullscreen when changing orientation");
+
+  promise_test(async (t) => {
     await test_driver.bless("request full screen");
     await document.documentElement.requestFullscreen();
-    const lockPromise = iframe.contentWindow.screen.orientation.lock(
+    const { orientation } = screen;
+    const oppositeOrientation = getOppositeOrientation();
+    const lockPromise = orientation.lock(oppositeOrientation);
+    const fsExitPromise = document.exitFullscreen();
+    await promise_rejects_dom(t, "AbortError", lockPromise);
+    await fsExitPromise;
+  }, "Exiting fullscreen causes a pending lock to be aborted");
+
+  promise_test(async (t) => {
+    const iframe = await attachIframe();
+    t.add_cleanup(makeCleanup(iframe));
+    const iframeWindow = iframe.contentWindow;
+    await test_driver.bless("request full screen", null, iframeWindow);
+    await iframe.contentDocument.documentElement.requestFullscreen();
+    const lockPromise = iframeWindow.screen.orientation.lock(
       getOppositeOrientation()
     );
-    await document.exitFullscreen();
-    const frameDOMException = iframe.contentWindow.DOMException;
-    await promise_rejects_dom(t, "AbortError", frameDOMException, lockPromise);
-  }, "Fully unlocking the screen orientation causes a pending lock in a nested browsing context to be aborted");
+    const fsExitPromise = iframe.contentDocument.exitFullscreen();
+    await promise_rejects_dom(
+      t,
+      "AbortError",
+      iframeWindow.DOMException,
+      lockPromise
+    );
+    await fsExitPromise;
+  }, "Exiting fullscreen from an iframe causes a pending lock to be aborted");
+
+  // promise_test(async (t) => {
```

Comment:
NB: I commented this test out on purpose. This test would work if `orientationManager->unlock()` would work. That should cause the orientation to change and the event to fire.

Reply:
Doesn't seem right to keep a test commented out. Will the test be commented out upstream as well? We can't diverge from upstream.

Reply:
Nah, was just temporary as I was trying to figure out why calling `orientationManager->unlock()` doesn't work.